frameworks/kafka/README.md

```diff
@@ -1,14 +1,420 @@
+
+### Resizing a Node
+The CPU and Memory requirements of each node may be increased or decreased as follows:
+- CPU (1.0 = 1 core): `BROKER_CPUS`
+- Memory (in MB): `BROKER_MEM` 
+
+Note that volume requirements (type and/or size) may not be changed after initial deployment.
+
+### Updating Placement Constraints
+
+Placement constraints may be updated after initial deployment using the following procedure. See [Service Settings](#service-settings) above for more information on placement constraints.
+
+Let's say we have the following deployment of our nodes
+
+- Placement constraint of: `hostname:LIKE:10.0.10.3|10.0.10.8|10.0.10.26|10.0.10.28|10.0.10.84`
+- Tasks:
+```
+10.0.10.3: _NODEPOD_-0
+10.0.10.8: _NODEPOD_-1
+10.0.10.26: _NODEPOD_-2
+10.0.10.28: empty
+10.0.10.84: empty
+```
+
+`10.0.10.8` is being decommissioned and we should move away from it. Steps:
+
+1. Remove the decommissioned IP and add a new IP to the placement rule whitelist by editing `NODE_PLACEMENT`:
+
+	```
+	hostname:LIKE:10.0.10.3|10.0.10.26|10.0.10.28|10.0.10.84|10.0.10.123
+	```
+1. Redeploy `_NODEPOD_-1` from the decommissioned node to somewhere within the new whitelist: `dcos kafka pods replace _NODEPOD_-1`
+1. Wait for `_NODEPOD_-1` to be up and healthy before continuing with any other replacement operations.
+
+
+## Restarting nodes
+
+This operation will restart a node, while keeping it at its current location and with its current persistent volume data. This may be thought of as similar to restarting a system process, but it also deletes any data which isn't in a persistent volume, via the magic of containers.
+
+1. Run `dcos kafka pods restart kafka-0`
+
+## Replacing nodes
+
+This operation will move a node to a new system, and will discard the persistent volumes at the prior system to be rebuilt at the new system. Perform this operation if a given system is about to be offlined or has already been offlined. Note that nodes are not moved automatically; you must manually perform the following steps to move tasks to new systems. You may build your own automation to perform node replacement automatically according to your own preferences.
+
+1. _ANY STEPS TO WIND DOWN A NODE BEFORE IT'S WIPED/DECOMMISSIONED GO HERE_
+1. Run `dcos kafka pods replace kafka-0` to halt the current instance (if still running) and launch a new instance elsewhere.
+
+For example, let's say `kafka-0`'s host system has died and `kafka-0` needs to be moved.
+
+	```
+	dcos kafka pods replace _NODEPOD_-3
+	```
+
+<a name="disaster-recovery"></a>
+# Disaster Recovery
+
+## Backup/Restore
+  For more information on Apache Kafka backup/restore, please see its [documentation](http://kafka.apache.org/documentation.html)
```

Comment:
Maybe add something like "The DC/OS Apache Kafka does not natively support any backup or restore functionality. Backup and restore are critical to any disaster recovery plan."